src/db/models.rs

```diff
@@ -0,0 +1,60 @@
+use teloxide::types::{ChatId, MessageId};
+
+/// Represents a link between chat rooms: a user in private messages and a forum topic.
+/// The values of the `sender_chat` i `recipient_chat` fields can be swapped.
+/// When receiving the structure, `sender_chat` is the initiator.
+/// The `last_private`, `last_topic` fields store the id of the last message in a private and a topic respectively.
+#[derive(Clone, Copy, Debug)]
+pub struct MappingChat {
+    pub sender_chat: ChatId,
+    pub recipient_chat: ChatId,
+    pub last_private: MessageId,
+    pub last_topic: MessageId,
+}
+
+impl MappingChat {
+    pub fn new(
+        private_chat: ChatId, 
+        forum_topic: ChatId, 
+        last_private: MessageId, 
+        last_topic: MessageId,
+    ) -> Self {
+        Self {
+            sender_chat: private_chat,
+            recipient_chat: forum_topic,
+            last_private,
+            last_topic,
+        }
+    }
+
+    pub fn sync(&mut self, last_private: MessageId, last_topic: MessageId) {
+        self.last_private = last_private;
+        self.last_topic = last_topic;
+    }
+
+    pub fn unique_id(&self) -> i64 {
+        self.sender_chat.0.min(self.recipient_chat.0)
+    }
+}
+
+impl From<(i64, i64, i32, i32)> for MappingChat {
```

Comment:
I recommend moving out such tuples into the `type` declaration. Yes, it is still just a tuple, but it will be hidden under a meaningful name with an appropriate description. And you will not need to copy-paste this tuple. It will improve the code's readability and understandability.